tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ProcessStart/ProcessStartIntegration.cs

```diff
@@ -0,0 +1,59 @@
+// <copyright file="ProcessStartIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Process
+{
+    /// <summary>
+    /// System.Net.Http.HttpClientHandler calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+       AssemblyNames = new[] { "System", "System.Diagnostics.Process" },
+       TypeName = "System.Diagnostics.Process",
+       MethodName = "Start",
+       ReturnTypeName = ClrNames.Process,
+       MinimumVersion = "1.0.0",
+       MaximumVersion = "7.*.*",
+       IntegrationName = nameof(Configuration.IntegrationId.ProcessStart))]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ProcessStartIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance)
+        {
+            if (instance is System.Diagnostics.Process process)
+            {
+                return new CallTargetState(scope: ProcessStartCommon.CreateScope(process.StartInfo));
+            }
```

Comment:
You might be able to avoid null-checking the StartInfo property later by doing the following syntax. I guess it's subjective as to which syntax is clearer and preferred

```suggestion
            if (instance is System.Diagnostics.Process { StartInfo: { } startInfo })
            {
                return new CallTargetState(scope: ProcessStartCommon.CreateScope(startInfo));
            }
```

Reply:
I suppose you could also do the following which is longer but maybe more clear:

```suggestion
            if (instance is System.Diagnostics.Process process && process.StartInfo is var startInfo)
            {
                return new CallTargetState(scope: ProcessStartCommon.CreateScope(startInfo));
            }
```

Reply:
Cool. Thanks!